### DISCUSSION TITLE
Programmatically using Qualifiers

### DISCUSSION THREAD

**Discussion by rsenden** [micronaut-projects/micronaut-core#6338] [Q&A] [answered]
I cannot seem to find any examples on how to programmatically use qualifiers to get particular beans/bean definitions.

As an example, suppose I have an annotation like the following:
```java
@Qualifier
@Retention(RetentionPolicy.RUNTIME)
public @interface SubcommandOf {
	Class<?> value();
}
```

How can I define qualifiers to have `ApplicationContext#getBeanDefinitions(Qualifier)` return the following:
1. All bean definitions for classes annotated with `@SubcommandOf` (not taking into account the actual annotation value)
2. All bean definitions for classes with a particular annotation value, for example `@SubcommandOf(SomeParticularClass.class)`

Note: I'm new to Micronaut, but have extensive experience with Spring DI.

**Comment by graemerocher** [comment, marked as answer]
1. Is easy. Use `Qualifiers.byStereotype(SubcommandOf.class)`
2. The second one it seems we don't really have a programmatic API though you can always filter the results of the previous call.

**Reply by rsenden** [reply]
Thanks, `Qualifiers.byStereoType(SubcommandOf.class)` works perfectly.
